crates/uv-resolver/src/lock.rs

```diff
@@ -586,16 +586,15 @@ impl Distribution {
                         url: VerbatimUrl::from_url(url),
                     };
                     let built_dist = BuiltDist::DirectUrl(direct_dist);
-                    Dist::Built(built_dist)
-                }
-                SourceKind::Git(_) => {
-                    unreachable!("Wheels cannot come from Git sources")
```

Comment:
I think the original intent here was that these cases were actually impossible as a contract of constructing a `Distribution`. I wonder if that got flubbed somehow.

Reply:
Yeah this should be impossible.

Reply:
If I can't make the types do it for me, then I usually define a `TryFrom` impl and point Serde to that. For example: https://github.com/astral-sh/uv/blob/04c4da4e65d61347a19960f1ba94cc26d4d01458/crates/uv-resolver/src/lock.rs#L333